File: Contracts/PaymentRequestDto.cs
using System.ComponentModel.DataAnnotations;

namespace Contracts;

public class PaymentRequestDto
{
    public PaymentRequestDto(int month, double amount, int year, int subsId)
    {
        Month = month;
        Amount = amount;
        Year = year;
        SubsId = subsId;
    }

    public string UserId { get; set; } = null!;

    [Required]
    [MinLength(13)]
    [MaxLength(16)]
    public string CardNumber { get; set; } = null!;
    [Required]
    [Range(1,12)]
    public int Month { get; set; }
    [Range(0, double.MaxValue)]
    public double Amount { get; set; }
    [Required]
    public int Year { get; set; }

    [Required]
    [MinLength(3)]
    [MaxLength(3)]
    public string Code { get; set; } = null!;
    public int SubsId { get; set; }
}

File: mobile/lib/dto/subscription/payment_request_dto.dart
class PaymentRequestDto {
  late String userId;
  final String cardNumber;
  final int month;
  final double amount;
  final int year;
  final String code;
  final int subsId;


  PaymentRequestDto({
    required this.userId,
    required this.cardNumber,
    required this.month,
    required this.amount,
    required this.year,
    required this.code,
    required this.subsId
  });


  factory PaymentRequestDto.fromJson(dynamic json)
  {
    return PaymentRequestDto(
        userId: "userId",
        cardNumber: json["cardNumber"],
        month: json["month"],
        amount: json["amount"],
        year: json["year"],
        code: json["code"],
        subsId: json["subsId"]
    );
  }
}

